src/autodiff.jl

```diff
@@ -122,130 +102,34 @@ struct StateJacobian{VI, VT, MT, SMT, VP, VD, SubT, SubD} <: AbstractJacobian
         Vre = Float64.([i for i in 1:n])
         Vim = Float64.([i for i in n+1:2*n])
         V = Vre .+ 1im .* Vim
-        J = residual_jacobian(V, Y, pv, pq)
+        J = structure.sparsity(V, Y, ref, pv, pq)
```

Comment:
Good point. 
I am wondering if we should not pass directly the matrix `J` as input for the sparsity detection phase. I am afraid that at the moment we are not completely generic, as we could not infer a priori the signature of the function `structure.sparsity`

Reply:
I agree. I should look into that. I put it on my list. That's nearly there.